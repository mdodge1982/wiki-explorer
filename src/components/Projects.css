.header {
	padding: 1em 2em .5em;
	margin: 0;
	position: fixed;
	top: 0;
	width: 100%;
	background: linear-gradient(to bottom, #fff 0%,#f7f7f7 100%);
	border-bottom: 1px solid #A9D8F7;
}
.header a {
	display: inline-block;
	padding: .2em .3em;
}
.Projects {
	padding: 8em 2em;
	border-bottom: none;
	list-style-type: none;
}
