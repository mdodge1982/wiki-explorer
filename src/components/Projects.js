import React, {Component} from 'react';
import {connect} from 'react-redux';
import './Projects.css';
import {listentoSocket,fetchProjects} from '../actions';
import Project from './Project';
import PageDetail from './PageDetail';

class ProjectsComponent extends Component {
	constructor(props) {
		super(props);
		props.dispatch(listentoSocket());
		props.dispatch(fetchProjects());
	}

	render() {
		//Create an index for navigating the long list of projects
		const idxArr = [];
		let curIdx = '';
		//Create the li's
		const projects = this.props.allNames.map(projectName => {
			let isNewIdx = false;
			if(projectName.charAt(0)!==curIdx){
				//New letter for the index
				isNewIdx = true;
				curIdx = projectName.charAt(0);
				idxArr.push(curIdx);
			}
			return (
				//Tell the Project instance whether it needs to add an anchor for the index
				<Project key={projectName} projectName={projectName} isNewIdx={isNewIdx}/>
			);
		});
		return (
			<div>
				<div className="header">
					<p className="right mobile-hide"><small>Front-end code by Meredith Dodge</small></p>
					<h1>Wiki Meta Explorer</h1>
					<a href="#top" className="right">Top</a>
					<div>{idxArr.map(letter => (
						<a key={letter} href={'#'+letter}>{letter}</a>
					))}</div>
				</div>
				<ul className="Projects jump-link" id="top">
					{projects}
				</ul>
				<PageDetail />
			</div>
		);
	}
}

//Use connect to get state and dispatch
const Projects = connect(({allNames}) => ({allNames}))(ProjectsComponent);

export default Projects;
