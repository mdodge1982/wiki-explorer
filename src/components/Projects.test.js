import React from 'react';
import Projects from './Projects';
import {Provider} from 'react-redux';
import store from '../store';
import renderer from 'react-test-renderer';

jest.mock('../actions/websocket');

test('Test render', () => {
	store.dispatch({
		type: 'PROJECT.LIST',
		data: ['My Fun Project']
	});
	const component = renderer.create(
		<Provider store={store}>
			<Projects />
		</Provider>
	);
	let tree = component.toJSON();
	expect(tree).toMatchSnapshot();

});
