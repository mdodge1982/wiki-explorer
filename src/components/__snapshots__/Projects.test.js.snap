// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test render 1`] = `
<div>
  <div
    className="header"
  >
    <p
      className="right mobile-hide"
    >
      <small>
        Front-end code by Meredith Dodge
      </small>
    </p>
    <h1>
      Wiki Meta Explorer
    </h1>
    <a
      className="right"
      href="#top"
    >
      Top
    </a>
    <div>
      <a
        href="#M"
      >
        M
      </a>
    </div>
  </div>
  <ul
    className="Projects jump-link"
    id="top"
  >
    <li
      className="Project"
    >
      <span
        className="named-anchor"
        id="M"
      />
      <h4
        onClick={[Function]}
      >
        My Fun Project
         
        
      </h4>
      
    </li>
  </ul>
  
</div>
`;
